modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/open_storage_query.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2023 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Peripherals
+    module StorageInteraction
+      module Nextcloud
+        class OpenStorageQuery
+          def initialize(storage)
+            @uri = storage.uri
+          end
+
+          def self.call(storage:, user:)
+            new(storage).call(user:)
+          end
+
+          # rubocop:disable Lint/UnusedMethodArgument
+          def call(user:)
+            ServiceResult.success(result: File.join(@uri.to_s, 'index.php/apps/files'))
```

Comment:
Could it be just like that without path? If no, I'd suggest to use `Storages::Peripherals::StorageInteraction::Nextcloud::Util.join_uri_path`. I believe we agreed to use this one instead of `File.join`.
```suggestion
            ServiceResult.success(result: @uri.to_s)
```

Reply:
without the path it opens the dashboard. some time ago, we agreed on the requirement, that the root open link should point to files app.

Reply:
Okay, I see, thanks!
What about using `Storages::Peripherals::StorageInteraction::Nextcloud::Util.join_uri_path`?

Reply:
Yes, changed that, thanks for noticing!